File: language/c++/69-sqrt-x.cpp
class Solution {
public:
    int mySqrt(int x) {
        double res = x / 2.0;
        while( abs(res*res - x) > 0.0001) {
            res = res - (res*res - x) / (2.0 * res);
        }
        return (int)res;
    }
};

File: language/python/69-sqrt-x.py
class Solution:
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        ans = x / 2.0
        while abs(ans*ans - x) > 0.00001:
            ans = ans - (ans*ans - x) / (2.0*ans)
        return int(ans)
        
